# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentLoadListener.cpp
# Commit: bb05fe8fa742
# Full Hash: bb05fe8fa742e7e470509f52a957d3b82ced2acb
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-06-04 10:21:11
# Description:
#   Bug 1713365, don't try to bfcache if the session history entry already has a frameloader, r=peterv
#   
#   The crash happens currently when there are more than one loading session history entries at the same time and when
#   the latter load then accesses current active entry, it has already the bfcached frameloader.
#   
# ==============================================================================

diff -r 52ed1b7f1fb0 -r bb05fe8fa742 netwerk/ipc/DocumentLoadListener.cpp
--- a/netwerk/ipc/DocumentLoadListener.cpp	Thu Jun 03 20:31:22 2021 +0000
+++ b/netwerk/ipc/DocumentLoadListener.cpp	Thu Jun 03 20:34:24 2021 +0000
@@ -1713,6 +1713,8 @@
         browsingContext->AllowedInBFCache(mDocumentChannelId);
     if (options.mTryUseBFCache) {
       options.mReplaceBrowsingContext = true;
+      options.mActiveSessionHistoryEntry =
+          browsingContext->GetActiveSessionHistoryEntry();
     }
   }
 
